client/components/jetpack-header/test/index.js

```diff
@@ -2,44 +2,53 @@
  * @jest-environment jsdom
  */
 
-import { shallow, mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import { JetpackHeader } from '..';
 
+jest.mock( 'calypso/components/jetpack-logo', () => ( { size } ) => (
+	<div data-testid="jetpack-logo">
+		<div data-testid="jetpack-logo-size">{ size }</div>
+	</div>
+) );
+jest.mock( '../partner-logo-group', () => ( { width } ) => (
+	<div data-testid="jetpack-partner-logo-group">
+		<div data-testid="jetpack-partner-logo-group-size">{ width }</div>
+	</div>
+) );
+
 describe( 'JetpackHeader', () => {
 	test( 'renders default Jetpack logo when no partnerSlug prop', () => {
-		const wrapper = shallow( <JetpackHeader /> );
-		expect( wrapper.children() ).toHaveLength( 1 );
-		expect( wrapper.children().name() ).toEqual( 'JetpackLogo' );
-		expect( wrapper.find( 'JetpackLogo' ).props().size ).toEqual( 45 );
+		render( <JetpackHeader /> );
+		expect( screen.queryByTestId( 'jetpack-logo' ) ).toBeVisible();
+		expect( screen.queryByTestId( 'jetpack-logo-size' ) ).toHaveTextContent( '45' );
 	} );
 
 	test( 'should render default Jetpack logo when passed empty string for partnerSlug prop', () => {
-		const wrapper = shallow( <JetpackHeader partnerSlug="" /> );
-		expect( wrapper.children() ).toHaveLength( 1 );
-		expect( wrapper.children().name() ).toEqual( 'JetpackLogo' );
-		expect( wrapper.find( 'JetpackLogo' ).props().size ).toEqual( 45 );
+		render( <JetpackHeader partnerSlug="" /> );
+		expect( screen.queryByTestId( 'jetpack-logo' ) ).toBeVisible();
+		expect( screen.queryByTestId( 'jetpack-logo-size' ) ).toHaveTextContent( '45' );
```

Comment:
This kinda just seems to be testing the mock and that a value is passed to its props -- I'm wondering how useful that really is 🤔

Reply:
TBH It's a compromise. While `testing-library` recommends that we test what the user sees (and avoid mocks), if we go with that approach, we step over the test responsibility of the other component. So we're making a decision here: should we test ONLY this component's behavior (which is passing props to the child component), or do it more like an integration test and test details from the other component. The decision here is to avoid mixing responsibilities, but at some point we might want to head in the other direction, which is what we're doing in Gutenberg.

Reply:
Using RTL I'm (almost) always in favor of more integrated tests and testing what props are passed on doesn't always yield a lot of value imho. As Marin said it's a compromise of testing responsibilities.

Reply:
Yeah, that makes sense. It definitely seems challenging to find that balance! I agree about liking integrated tests, since most things are used in an integrated environment anyways. 👍